src/App.js

```diff
@@ -2,29 +2,57 @@ import React, { useState } from "react";
 import "./App.css";
 import Header from "./components/UI/Header";
 import Card from "./components/UI/Card";
-import Caption from "./components/UI/Caption";
-import StyleChanger from "./components/StyleChanger/StyleChanger";
+import Caption from "./components/Caption/Caption";
 
 function App() {
+  var text = `Building interactive user interfaces in React is fun and easy. You
+  just need to describe how parts of the application interface look in
+  different states. React will update them in a timely manner when the
+  data changes`;
+
   const [cbValue, setCbValue] = useState("");
+  const [penClicked, setPenClicked] = useState(false);
+  const [currentTextValue, setCurrentTextValue] = useState(text);
+  const [lastTextValue, setLastTextValue] = useState(text);
 
   const cbValueHandler = (val) => {
     setCbValue(val);
   };
 
+  const penClickedHandler = (val) => {
+    setPenClicked(val);
+  };
+
+  const saveClickHander = () => {
+    setLastTextValue(currentTextValue);
+  };
+
+  const cancelClickHandler = () => {
+    setCurrentTextValue(lastTextValue);
+  };
+
+  const inputFinishHandler = (event) => {
+    setCurrentTextValue(event.target.innerText);
+  };
+
```

Comment:
Same here. These methods are low level actions overs card elements. So should be at least placed in Card component.